lib/odata/query/result.rb

```diff
@@ -52,7 +61,16 @@ def next_page
       end
 
       def next_page_url
-        next_page.attributes['href'].value.gsub(service.service_url, '')
+        return unless next_page && next_page.attributes['href']
```

Comment:
Maybe let's save the output of `next_page` to avoid parsing the same response 3 times?

Reply:
I believe next_page can change after the code runs `service.execute`. So if we cache it, we won't get to the next page.

Reply:
It may also not matter for our use-case with Chocolatey, but I would prefer not to break it more than we have to

Reply:
I mean "cache" in this method only. Something like:
```
      def next_page_url
        next_page_value = next_page
        return unless next_page_value && next_page_value.attributes['href']

        # We used to get the url in http format, then it changed
        # to https. Let's remove both
        http_verison =  service.service_url.sub('https://', 'http://')
        https_version = service.service_url.sub('http://', 'https://')
        next_page_value.attributes['href']
                 .value
                 .gsub(http_verison, '')
                 .gsub(https_version, '')
      end
```